skimage/metrics/simple_metrics.py

```diff
@@ -158,3 +161,99 @@ def peak_signal_noise_ratio(image_true, image_test, *, data_range=None):
 
     err = mean_squared_error(image_true, image_test)
     return 10 * np.log10((data_range ** 2) / err)
+
+
+def _pad_to(arr, shape):
+    """Pad an array with trailing zeros to a given target shape.
+
+    Parameters
+    ----------
+    arr : ndarray
+        The input array.
+    shape : tuple
+        The target shape.
+
+    Returns
+    -------
+    padded : ndarray
+        The padded array.
+
+    Examples
+    --------
+    >>> _pad_to(np.ones((1, 1), dtype=int), (1, 3))
+    array([[1, 0, 0]])
+    """
+    if not all(s >= i for s, i in zip(shape, arr.shape)):
+        raise ValueError("Target shape must be strictly greater "
+                         "than input shape.")
+    padding = [(0, s-i) for s, i in zip(shape, arr.shape)]
+    return np.pad(arr, pad_width=padding, mode='constant', constant_values=0)
+
+
+def normalized_mutual_information(im_true, im_test, *, bins=100):
+    r"""Compute the normalized mutual information.
+
+    The normalized mutual information is given by::
+
+    ..math::
+        Y(A, B) = \frac{H(A) + H(B)}{H(A, B)}
+
+    where :math:`H(X)` is the entropy,
+    :math:`- \sum_{x \in X}{x \log x}.`
+
+    It was proposed to be useful in registering images by Colin Studholme and
+    colleagues [1]_. It ranges from 1 (perfectly uncorrelated image values)
+    to 2 (perfectly correlated image values, whether positively or negatively).
+
+    Parameters
+    ----------
+    im_true, im_test : ndarray
+        Images to be compared. The two input images must have the same number
+        of dimensions.
+    bins : int or sequence of int, optional
+        The number of bins along each axis of the joint histogram.
+
+    Returns
+    -------
+    nmi : float
+        The normalized mutual information between the two arrays, computed at
+        the granularity given by ``bins``. Higher NMI implies more similar
+        input images.
+
+    Raises
+    ------
+    ValueError
+        If the images don't have the same number of dimensions.
+
+    Notes
+    -----
+    If the two input images are not the same shape, the smaller image is padded
+    with zeros.
+
+    References
+    ----------
+    .. [1] C. Studholme, D.L.G. Hill, & D.J. Hawkes (1999). An overlap
```

Comment:
add wikipedia link for mutual information?

Reply:
```suggestion
    .. https://en.wikipedia.org/wiki/Mutual_information
    .. [2] C. Studholme, D.L.G. Hill, & D.J. Hawkes (1999). An overlap
```